api/Insight_API_documentation.md

```diff
@@ -0,0 +1,1200 @@
+               "spentHeight": null
+            },
+            {
+               "value": 95.87354,
+               "n": 3,
+               "scriptPubKey": {
+                  "hex": "bb76a914b7a0fadc16c16c1d6607b3e8ebc3ff4fc214a87488ac",
+                  "asm": "OP_SSGEN OP_DUP OP_HASH160 b7a0fadc16c16c1d6607b3e8ebc3ff4fc214a874 OP_EQUALVERIFY OP_CHECKSIG",
+                  "addresses": [
+                     "DshhqzchyKUzPuZ8QQWwuepCg7qzBd8yUrt"
+                  ],
+                  "type": "stakegen"
+               },
+               "spentTxId": null,
+               "spentIndex": null,
+               "spentHeight": null
+            }
+         ],
+         "blockhash": "0000000000000000bffc01c7350dd0a58619e340ab4d75d8d3ff42e201d67503",
+         "blockheight": 272308,
+         "confirmations": 7501,
+         "time": 1536307801,
+         "blocktime": 1536307801,
+         "valueOut": 95.885506,
+         "size": 420,
+         "valueIn": 95.885506,
+         "fees": 1.0E-8
+      }
+   ]
+}
+```
+
+<br/>
+
+## Status and Utility
+
+Methods that provide utilities or relay network status.
+
+### /sync/
+
+**URL:**  ```GET /sync/```
+
+**Description:** Syncs with historical blockchain data. **REVIEW WORDING**
+
+
+**Request Example:**
+
+```GET /sync```
+
+**Request Response:**
+
+```
+{
+    "status": "finished",
+    "blockChainHeight": 279815,
+    "syncPercentage": 100,
+    "height": 279815,
+    "error": null,
+    "type": "from RPC calls"
+}
+```
+
+<br/>
+
+### /peer/
+
+**URL:**  ```GET /peer/```
+
+**Description:** Retrieves Peer-to-Peer (P2P) data sync status.  **REVIEW WORDING**
+
+**Request Example:**
+
+```GET /peer/ ```
+
+**Request Response:**
+
+```
+{
+    "connected": true,
+    "host": "127.0.0.1",
+    "port": null
+}
+```
+
+<br/>
+
+### /status/
+
+**URL:**  ```GET /status?q=xxx```
+
+**Description:** Retrieves status of Decred network. If `q` is set to a parameter from the table below, only that parameter is returned. If `q` is not specified, all status parameters are returned. 
+
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| q              | `string`      |   (optional) getInfo </br> (optional) getDifficulty </br> (optional) getBestBlockHash </br> (optional) getLastBlockHash </br>|
+
+**Request Example:**
+
+```
+GET /status?q=getDifficulty 
+```
+
+**Request Response:**
+
+```
+{
+    "difficulty": 6484304315.871729
+}
+```
+
+<br/>
+
+### /estimatefee/
+
+**URL:**  ```GET /status?nbBlocks=N```
+
+**Description:** Retrieves estimate of transaction fee. **REVIEW**
+
+**Parameters:**
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| N             | `int64`      |   (optional) Number of blocks **REVIEW. What does nbBlocks mean here?**    |  
+
```

Comment:
good question.  I think "number of blocks" in the future that you want your transaction mined in.  We always return 0.001 right now.

Reply:
Leaving out the 0.001 for now, as it's always dangerous to document hard-coded values that might change, as the docs might not get updated in a timely manner (or at all). Going with the below wording. Is this still correct? 
`(optional) Number of blocks into the future that the transaction can be mined in (validated) `

Reply:
I suggest going with the description closer to what's at the above link.

"Retrieves an estimate the required transaction fee for a transaction to be included within a certain number of blocks"

And the description of the N parameter similarly:

"The number of block within which the transaction should be mined"

Reply:
Currently going with this (in local branch, not pushed yet):

Description:
```Retrieves an estimate of the fee required for a transaction to be included within a certain number of blocks (nbBlocks). If `nbBlocks` is not specified, it defaults to 2.```

Parameter Description:
```(optional) Number of blocks within which the transaction should be mined```

Reply:
@papacarp , good catch on the utils/estimatefee URL being wrong. Corrected in last changes push.

Reply:
`nbBlocks` for the parameter instead of `N`